client/core/trade.go

```diff
@@ -2844,24 +2830,50 @@ func (c *Core) sendRedeemAsync(t *trackedTrade, match *matchTracker, coinID, sec
 	}()
 }
 
+func (t *trackedTrade) redeemFee() uint64 {
+	// Try not to use (*Core).feeSuggestion here, since it can incur an RPC
+	// request to the server. t.redeemFeeSuggestion is updated every tick and
+	// uses a rate directly from our wallet, if available. Only go looking for
+	// one if we don't have one cached.
+	var feeSuggestion uint64
+	if _, is := t.accountRedeemer(); is {
+		feeSuggestion = t.metaData.RedeemMaxFeeRate
+	} else {
+		feeSuggestion = t.redeemFeeSuggestion.get()
+	}
+	if feeSuggestion == 0 {
+		feeSuggestion = t.dc.bestBookFeeSuggestion(t.wallets.toWallet.AssetID)
+	}
+	return feeSuggestion
+}
+
 // confirmRedemption checks if the user's redemption has been confirmed,
 // and if so, updates the match's status to MatchConfirmed.
 //
 // This method accesses match fields and MUST be called with the trackedTrade
 // mutex lock held for writes.
 func (t *trackedTrade) confirmRedemption(match *matchTracker) {
-	confirmer, isConfirmer := t.wallets.toWallet.Wallet.(asset.RedemptionConfirmer)
-	if !isConfirmer {
-		match.Status = order.MatchConfirmed
-		err := t.db.UpdateMatch(&match.MetaMatch)
-		if err != nil {
-			t.dc.log.Errorf("Failed to update match in db %v", err)
-		}
+	// In some cases the wallet will need to send a new redeem transaction.
+	toWallet := t.wallets.toWallet
+
+	if toWallet.peerCount < 1 {
+		t.dc.log.Errorf("Unable to confirm redemption. %s wallet has no peers.", unbip(toWallet.AssetID))
+		return
+	}
+
+	if !toWallet.synchronized() {
+		t.dc.log.Errorf("Unable to confirm redemption. %s still syncing.", unbip(toWallet.AssetID))
 		return
 	}
-	if !t.wallets.toWallet.connected() {
+
+	didUnlock, err := toWallet.refreshUnlock()
+	if err != nil { // Just log it and try anyway.
+		t.dc.log.Errorf("refreshUnlock error checking redeem %s: %v", t.wallets.toWallet.Symbol, err)
 		return
```

Comment:
This return is new. I think correct to return if unable to unlock? Maybe not necessary though if the redeem only needs to check confirms and not send a new transaction, which is most of the time. But I think wallet is expected to be unlocked with active trades anyhow.

Reply:
Why not let the action requiring an unlocked wallet throw the error then?

Reply:
Ok. Will remove the return then.

Reply:
Ok.  I mean you could still try to `refreshUnlock` here but if there's something downstream that really needed it to succeed, it can bomb.